File: SDKs/amazonaws.com/kinesisanalytics/2015-08-14/python/src/sdk/models/shared/s3referencedatasourcedescription.py
import dataclasses
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclasses.dataclass
class S3ReferenceDataSourceDescription:
    r"""S3ReferenceDataSourceDescription
    Provides the bucket name and object key name that stores the reference data.
    """
    
    bucket_arn: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('BucketARN') }})
    file_key: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('FileKey') }})
    reference_role_arn: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('ReferenceRoleARN') }})
    

File: SDKs/amazonaws.com/kinesisanalytics/2015-08-14/typescript/src/sdk/models/shared/s3referencedatasourcedescription.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



// S3ReferenceDataSourceDescription
/** 
 * Provides the bucket name and object key name that stores the reference data.
**/
export class S3ReferenceDataSourceDescription extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=BucketARN" })
  bucketARN: string;

  @SpeakeasyMetadata({ data: "json, name=FileKey" })
  fileKey: string;

  @SpeakeasyMetadata({ data: "json, name=ReferenceRoleARN" })
  referenceRoleARN: string;
}
